controllers/idler/idler_controller.go

```diff
@@ -56,7 +56,7 @@ type Reconciler struct {
 	AllNamespacesClient client.Client
 	ScalesClient        scale.ScalesGetter
 	DynamicClient       dynamic.Interface
-	GetHostCluster      cluster.GetHostClusterFunc
+	GetHostCluster      cluster.GetClustersFunc
```

Comment:
The name of the parameter is `GetHostCluster` in a singular form, but the function can return 0..n objects. Let's either change the name or the type of the parameter